dvc/output/base.py

```diff
@@ -256,6 +256,9 @@ def save(self):
                 )
             return
 
+        if self.metric:
+            self.verify_metric()
+
```

Comment:
There is a
```
            if self.metric or self.plot:
                self.verify_metric()
```
a few lines above, under `if not self.use_cache`. How about we just move it outside of that if and check before that?

Reply:
Fixed